src/mono/wasm/runtime-test.js

```diff
@@ -102,51 +102,56 @@ if (typeof globalThis.performance === 'undefined') {
     }
 }
 
-var Module = {
-    no_global_exports: true,
-    mainScriptUrlOrBlob: "dotnet.js",
-    config: null,
-    configSrc: "./mono-config.json",
-    print: console.log,
-    printErr: console.error,
-    onConfigLoaded: function () {
-        if (!Module.config) {
-            console.error("Could not find ./mono-config.json. Cancelling run");
-            fail_exec(1);
-        }
-        // Have to set env vars here to enable setting MONO_LOG_LEVEL etc.
-        for (let variable in processedArguments.setenv) {
-            Module.config.environment_variables[variable] = processedArguments.setenv[variable];
-        }
+loadDotnet("./dotnet.js").then((createDotnetRuntime) => {
```

Comment:
Do we need to capture the promise result `{ MONO, INTERNAL, BINDING, Module }` for line 152?

Reply:
We pass it on line 134. 
BTW this file is still CommonJS loaded to global namespace but without global `Module`.
So it's bit ugly. Once I'm confident with ES6, I will switch this too and make it nicer.

Reply:
The whole `loadDotnet` will go away and with top level await `{ MONO, INTERNAL, BINDING, Module }` would be top level too.

Reply:
Cool, and thanks for this work! I was trying it out on deno runtime with your branch: https://github.com/am11/runtime/commit/8dea0bb01cb5b137bbbfa643d894e0e055c40ea5, will follow-up once we are fully switched to modules. :)

The Venn's diagram of support matrices between https://caniuse.com/?search=modules and https://caniuse.com/?search=wasm seem to agree that it's safe to switch dotnet.js to module encapsulation.